tests/slist_ut/slist_ut.c

```diff
@@ -0,0 +1,1114 @@
+    ASSERT_ARE_EQUAL(int, 0, result);
+
+    result = umocktypes_charptr_register_types();
+    ASSERT_ARE_EQUAL(int, 0, result);
+
+    REGISTER_UMOCK_ALIAS_TYPE(PSINGLYLINKEDLIST_ENTRY, void*);
+}
+
+TEST_SUITE_CLEANUP(TestClassCleanup)
+{
+    umock_c_deinit();
+    TEST_MUTEX_DESTROY(test_serialize_mutex);
+}
+
+TEST_FUNCTION_INITIALIZE(TestMethodInitialize)
+{
+    if (TEST_MUTEX_ACQUIRE(test_serialize_mutex))
+    {
+        ASSERT_FAIL("our mutex is ABANDONED. Failure in test framework");
+    }
+}
+
+TEST_FUNCTION_CLEANUP(TestMethodCleanup)
+{
+    umock_c_reset_all_calls();
+    TEST_MUTEX_RELEASE(test_serialize_mutex);
+}
+
+/*slist_initialize*/
+
+/* Tests_SRS_SLIST_07_001: [ If `list_head` is `NULL`, `slist_initialize` shall fail and return `false`. ]*/
+TEST_FUNCTION(slist_initialize_with_NULL_fails)
+{
+    // arrange
+
+    // act
+    bool result = slist_initialize(NULL);
+
+    // assert
+    ASSERT_ARE_EQUAL(bool, false, result);
+}
+
+/* Tests_SRS_SLIST_07_002: [ `slist_initialize` shall initialize the `next` pointer in `list_head` points to `NULL`and return `true` on success. ]*/
+TEST_FUNCTION(slist_initialize_succeeds)
+{
+    // arrange
+    SINGLYLINKEDLIST_ENTRY head;
+
+    // act
+    bool result = slist_initialize(&head);
+
+    // assert
+    ASSERT_ARE_EQUAL(void_ptr, NULL, head.next);
+    ASSERT_ARE_EQUAL(bool, true, result);
+}
+
+/*slist_is_empty*/
+
+/* Tests_SRS_SLIST_07_038: [ If `list_head` is `NULL`, `slist_is_empty` shall fail and return `true`. ]*/
+TEST_FUNCTION(slist_is_empty_with_NULL_list_fails)
+{
+    // arrange
+
+    // act
+    bool result = slist_is_empty(NULL);
+
+    // assert
+    ASSERT_ARE_EQUAL(bool, true, result);
+}
+
+/* Tests_SRS_SLIST_07_003: [ `slist_is_empty` shall return `true` if there is no `SINGLYLINKEDLIST_ENTRY` in this list. ]*/
+TEST_FUNCTION(slist_is_empty_with_empty_list_succeeds)
+{
+    // arrange
+    SINGLYLINKEDLIST_ENTRY head;
+    slist_initialize(&head);
+
+    // act
+    bool result = slist_is_empty(&head);
+
+    // assert
+    ASSERT_ARE_EQUAL(bool, true, result);
+}
+
+/* Tests_SRS_SLIST_07_004: [ `slist_is_empty` shall return `false` if there is one or more entris in the list. ]*/
+TEST_FUNCTION(slist_is_empty_with_one_entry_succeeds)
+{
+    // arrange
+    static simpleItem simp1 = { 1, { NULL } };
+
+    SINGLYLINKEDLIST_ENTRY head;
+    slist_initialize(&head);
+
+    slist_add(&head, &(simp1.link));
+
+    // act
+    bool result = slist_is_empty(&head);
+
+    // assert
+    ASSERT_ARE_EQUAL(bool, false, result);
+}
+
+/* Tests_SRS_SLIST_07_004: [ `slist_is_empty` shall return `false` if there is one or more entris in the list. ]*/
+TEST_FUNCTION(slist_is_empty_with_multiple_entries_succeeds)
+{
+    // arrange
+    static simpleItem simp1 = { 1, { NULL } };
+    static simpleItem simp2 = { 2, { NULL } };
+    static simpleItem simp3 = { 3, { NULL } };
+
+    SINGLYLINKEDLIST_ENTRY head;
+    slist_initialize(&head);
```

Comment:
>slist_initialize [](http://example.com/codeflow?start=4&length=16)

This has a retuned value.
When we ignore on purpose returned values we place a (void) in front of the call:

(void)slist_initialize(&head);

That also tells compilers that we're on purpose ignoring the returned value and there should be nothing to be done with it.

Reply:
Same for all the other places.

Reply:
done